src/dune_pkg/archive_driver.mli

```diff
@@ -0,0 +1,17 @@
+open Stdune
+
+(** Knows how to extract archives of a particular format *)
+type t
+
+(** Driver for tarballs, possibly compressed *)
+val tar : t
+
+(** Driver for zip files *)
+val zip : t
+
+(** Return the driver that can extract a file of a given name. The decision is
+    made based on the file's suffix. *)
+val choose_for_filename : string -> t option
+
+(** Extract an archive into a given target directory *)
```

Comment:
```suggestion
(** [extract t ~archive ~target] extracts an archive from the [archive] path into a given [target] directory. If the [target] path does not exist, it will create it. *)
```
Adding a bit of extra information is good (as it allows having it using ocaml-lsp without having to jump to the definition). This suggestion isn't formatted well, sorry about that.